### DISCUSSION TITLE
Moving to a single workflow?

### DISCUSSION THREAD

**Discussion by MaxGhenis** [PolicyEngine/policyengine-us#4955] [Ideas]
I recently worked on another project and Claude suggested putting all actions in a single workflow: https://github.com/MaxGhenis/pippy/blob/main/.github/workflows/ci_cd.yaml

Compared to our current approach of multiple workflow files, this reduces code duplication and triggers the right actions depending on if committing vs pushing.

Any disadvantages, other than a larger file? @nikhilwoodruff @anth-volk also applies to other repos.

**Comment by nikhilwoodruff** [comment]
Maybe I'm missing something but- that file has a shared trigger for both PR and push right? I can think of publishing actions as being something that's done on push but not PR (e.g. we wouldn't want to publish to PyPI on PRs)

**Comment by MaxGhenis** [comment]
Oops, it's missing the key line today. But you can add `if` statements to particular actions: https://github.com/MaxGhenis/pippy/issues/7

**Comment by anth-volk** [comment]
I see a lot of value in doing this, as there are cases where a PR might only warn for, e.g., linting, then actually fail on push, leading to confusion.

**Comment by MaxGhenis** [comment]
See https://github.com/PolicyEngine/policyengine-us-data/pull/13 for an example of this
